reactor-core/src/test/java21/reactor/core/scheduler/LoomBoundedElasticSchedulerTest.java

```diff
@@ -0,0 +1,267 @@
+package reactor.core.scheduler;
+
+import java.util.Queue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import reactor.core.Disposable;
+import reactor.test.util.RaceTestUtils;
+import reactor.util.concurrent.Queues;
+
+class LoomBoundedElasticSchedulerTest {
+
+	Scheduler scheduler;
+
+	@BeforeEach
+	void setup() {
+		scheduler = new LoomBoundedElasticScheduler(2,
+				3,
+				Thread.ofVirtual()
+				      .name("loom", 0)
+				      .factory());
+		scheduler.init();
+	}
+
+	@AfterEach
+	void teardown() {
+		scheduler.dispose();
+	}
+
+	@Test
+	public void ensuresTasksScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(1);
+
+		Disposable disposable = scheduler.schedule(latch::countDown);
+
+		Assertions.assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+		Assertions.assertThat(disposable.isDisposed()).isTrue();
+	}
+
+	@Test
+	public void ensuresTasksDelayedScheduling() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(1);
+
+		Disposable disposable = scheduler.schedule(latch::countDown, 200, TimeUnit.MILLISECONDS);
```

Comment:
200ms is actually quite a lot, can we go with, e.g. 5ms ? As this is an actual clock, not a dummy...